src/operator/subgraph/dnnl/dnnl_pow_mul_scalar.cc

```diff
@@ -0,0 +1,151 @@
+  bool scalar_is_int                 = param.exp_is_int && param.mul_is_int;
+  if (common::is_int(in_attrs->at(0)) && !scalar_is_int) {
+    TYPE_ASSIGN_CHECK(*out_attrs, 0, mshadow::kFloat64);
+  } else if (in_attrs->at(0) == mshadow::kBool) {
+    TYPE_ASSIGN_CHECK(*out_attrs, 0, scalar_is_int ? mshadow::kInt64 : mshadow::kFloat64);
+  } else {
+    TYPE_ASSIGN_CHECK(*out_attrs, 0, in_attrs->at(0));
+  }
+  return out_attrs->at(0) != -1;
+}
+
+inline static bool DNNLPowMulScalarStorageType(const nnvm::NodeAttrs& attrs,
+                                               const int dev_mask,
+                                               DispatchMode* dispatch_mode,
+                                               std::vector<int>* in_attrs,
+                                               std::vector<int>* out_attrs) {
+  return DNNLStorageType(attrs, dev_mask, true, dispatch_mode, in_attrs, out_attrs);
+}
+
+template <typename OP>
+static void ComputeOP(const nnvm::NodeAttrs& attrs,
+                      const OpContext& ctx,
+                      mshadow::Stream<cpu>* s,
+                      const TBlob& input,
+                      const TBlob& output,
+                      const double scalar) {
+  using namespace mshadow;
+  using namespace mshadow::expr;
+  MSHADOW_TYPE_SWITCH(output.type_flag_, DType, {
+    auto temp_req    = input.dptr_ == output.dptr_ ? kWriteInplace : kWriteTo;
+    TBlob temp_tblob = input;
+    if (input.type_flag_ != output.type_flag_) {
+      temp_tblob = TBlob(ctx.requested[0].get_space_typed<cpu, 1, DType>(Shape1(output.Size()), s));
+      CastCompute<cpu>(attrs, ctx, {input}, {kWriteTo}, {temp_tblob});
+    }
+    MXNET_ASSIGN_REQ_SWITCH(temp_req, Req, {
+      mxnet_op::Kernel<mxnet_op::op_with_req<OP, Req>, cpu>::Launch(
+          s, input.Size(), output.dptr<DType>(), temp_tblob.dptr<DType>(), DType(scalar));
+    });
+  });
+}
+
+static void PowMulScalarCompute(const nnvm::NodeAttrs& attrs,
+                                const OpContext& ctx,
+                                const std::vector<TBlob>& inputs,
+                                const std::vector<OpReqType>& req,
+                                const std::vector<TBlob>& outputs) {
+  mshadow::Stream<cpu>* s = ctx.get_stream<cpu>();
+  DCHECK_EQ(inputs.size(), 1);
+  DCHECK_EQ(outputs.size(), 1);
+  using namespace mshadow;
+  using namespace mshadow::expr;
+  const DNNLPowMulScalarParam& param = nnvm::get<DNNLPowMulScalarParam>(attrs.parsed);
+  // temp_mid_tblob is output of power operation and input of multiplication.
+  // Its dtype depends on input dtype and scalar type.
+  TBlob temp_mid_tblob =
+      ((common::is_int(inputs[0].type_flag_) || inputs[0].type_flag_ == kBool) &&
+       !param.exp_is_int) ?
+          outputs[0] :
+          inputs[0].type_flag_ == kBool ?
+          TBlob(ctx.requested[0].get_space_typed<cpu, 1, int64_t>(Shape1(inputs[0].Size()), s)) :
+          inputs[0];
```

Comment:
please use regular if else statements, it is not readable at all

Reply:
also are you sure that inputs[0] should be here as temp_mid_tblob?

Reply:
To me there is no "should". If the parameters of the operation are right (output of the power op is of the same dtype as the input) I believe it can be used as a temporary buffer.